sdcm/nemesis.py

```diff
@@ -4205,46 +4205,56 @@ def disrupt_add_remove_mv(self):
-                                              min_duration=300, max_duration=2400)
-            finally:
-                drop_materialized_view(session, ks_name, view_name)
+                    'Non-system keyspace and table are not found. nemesis can\'t be run')
+            ks_name, base_table_name = random.choice(ks_cfs).split('.')
+            view_name = f'{base_table_name}_view'
+            with self.cluster.cql_connection_patient(cql_query_executor_node) as session:
+                primary_key_columns = get_column_names(
+                    session=session, ks=ks_name, cf=base_table_name, is_primary_key=True)
+                # selecting a supported column for creating a materialized-view (not a collection type).
+                column = get_random_column_name(session=session, ks=ks_name,
+                                                cf=base_table_name, filter_out_collections=True)
+                if not column:
+                    raise UnsupportedNemesis(
+                        'A supported column for creating MV is not found. nemesis can\'t run')
+                column = f'"{column}"'
+                self.log.info("Stopping Scylla on node %s", self.target_node.name)
+                self.target_node.stop_scylla()
+                InfoEvent(message=f'Create a materialized-view for table {ks_name}.{base_table_name}').publish()
+                try:
+                    with EventsFilter(event_class=DatabaseLogEvent,
+                                      regex='.*view - Error applying view update.*',
+                                      extra_time_to_expiration=30):
+                        self.tester.create_materialized_view(ks_name, base_table_name, view_name, [column],
+                                                             primary_key_columns, session,
+                                                             mv_columns=[column] + primary_key_columns)
+                except Exception as error:  # pylint: disable=broad-except
+                    self.log.warning('Failed creating a materialized view: %s', error)
+                    self.target_node.start_scylla()
+                    raise
+                try:
+                    self.log.info("Starting Scylla on node %s", self.target_node.name)
+                    self.target_node.start_scylla()
+                    self.target_node.run_nodetool(sub_cmd="repair -pr")
+                    wait_for_view_to_be_built(self.target_node, ks_name, view_name, timeout=7200)
+                    session.execute(SimpleStatement(f'SELECT * FROM {ks_name}.{view_name} limit 1', fetch_size=10))
+                    sleep_for_percent_of_duration(self.tester.test_duration * 60, percent=1,
+                                                  min_duration=300, max_duration=2400)
+                finally:
+                    drop_materialized_view(session, ks_name, view_name)
+        finally:
+            self.unset_current_running_nemesis(cql_query_executor_node)
```

Comment:
is this unset really needed? isn't the default action of each nemesis to `unset_current_running_nemesis` when the call ends?

Reply:
so maybe i don't understand the initial problem here... why using `self.target_node` isn't good enough?

Reply:
if so, the 1st line here (that creates a `free_nodes` list) must be under a condition, like
`if self.target_node.running_nemesis:`

Reply:
We can't use the target node, since we want to do cql operation while it's down.

Target node is marked with nemesis before the disrupt function is even called, there's no point checking it.

Reply:
ah, i see, now i got it